packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { notificationService } from '@salesforce/salesforcedx-utils-vscode';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+
+/**
+ * Interface representing the metadata of a method.
+ */
+export interface MethodMetadata {
+  name: string;
+  parameters: Parameter[];
+  returnType: string;
+  isAuraEnabled: boolean;
+  className?: string;
+}
+
+/**
+ * Interface representing a parameter of a method.
+ */
+export interface Parameter {
+  name: string;
+  in: string;
+  required: boolean;
+  description: string;
+  schema: { type: string };
+}
+
+/**
+ * Class responsible for orchestrating metadata operations.
+ */
+export class MetadataOrchestrator {
+  /**
+   * Checks if a method is eligible for Apex Action creation.
+   * @param methodIdentifier - The identifier of the method.
+   * @returns True if the method is eligible, false otherwise.
+   */
+  public isMethodEligible = (methodIdentifier: string): boolean => {
+    // Placeholder for eligibility logic
+    return true;
+  };
+
+  /**
+   * Extracts metadata for the method at the current cursor position.
+   * @returns The metadata of the method, or undefined if no method is found.
+   */
+  public extractMethodMetadata = (): MethodMetadata | undefined => {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+      notificationService.showErrorMessage(nls.localize('no_active_editor'));
+      return;
+    }
+
+    const document = editor.document;
+    const cursorPosition = editor.selection.active;
+    const lines = document.getText().split('\n');
+    const currentLineIndex = cursorPosition.line;
+
+    let methodSignature = '';
+    let isAuraEnabled = false;
+
+    // Check if the preceding line contains @AuraEnabled
+    if (currentLineIndex > 0 && lines[currentLineIndex - 1].includes('@AuraEnabled')) {
+      isAuraEnabled = true;
+    }
+
+    // Traverse lines starting from the cursor position to construct the method signature
+    for (let line of lines) {
+      line = line.trim();
+      methodSignature += ` ${line}`;
+
+      // Stop once the closing parenthesis is reached
+      if (line.includes(') {')) {
+        break;
+      }
+    }
+
+    const methodMetadata = this.parseMethodSignature(methodSignature, isAuraEnabled);
+    if (!methodMetadata.isAuraEnabled) {
+      throw new Error(nls.localize('not_aura_enabled', methodMetadata.name));
+    }
+    return methodMetadata;
+  };
+
+  public extractAllMethodsMetadata = (): MethodMetadata[] | undefined => {
+    const editor = vscode.window.activeTextEditor;
```

Comment:
I think there is a use case that the active text editor is not available when we right click on one apex source file in the file explorer.